### PR TITLE
perf: methods comparison

### LINKED ISSUE #19: Optimization
Hello, 
First, thanks for this source code, it's nice :)
I think you should optimize method comparison in find function.

1. create an dictionary with methods names that map to a number:

```js
let mthMap = {
        GET: 1,
	HEAD: 2,	
	PATCH: 3,
	OPTIONS: 4,
	CONNECT: 5,
	DELETE: 6,
	TRACE: 7,
	POST: 8,
	PUT: 9,

	"": 0, // tip, this is ALL
}
```

2. add a mthidx to each routes:

```js
add(method, route, ...fns) {
	let { keys, pattern } = parse(route);
	let handlers = [].concat.apply([], fns);
	this.routes.push({ keys, pattern, method, handlers, 
              mthidx:mthMap[method]  // <<< here
        });
	return this;
}
```

3. in find, you can now compare numbers:

```js
find(method, url) {
	let mthidx = mthMap[method];
        let isHEAD = (mthidx === 2);

	let i=0, j=0, k, tmp, arr=this.routes;
	let matches=[], params={}, handlers=[];
	for (; i < arr.length; i++) {
		tmp = arr[i];
                // old code
                // if (tmp.method.length === 0 || tmp.method === method || isHEAD && tmp.method === 'GET') {

                // new code
		if (tmp.mthidx === mthidx  || !tmp.mthidx || (isHEAD && tmp.mthidx===1) ) {   
```

### PR DESCRIPTION
**Pull request by Kikobeats**
It's saving ~10ms

closes #19

### CODE REVIEW

File: index.mjs [hunk 1]
```diff
@@ -19,24 +32,25 @@ export default class Trouter {
...
 	find(method, url) {
-		let isHEAD=(method === 'HEAD');
+		let mthidx = mthMap[method];
+		let isHEAD = (mthidx === 2);
 		let i=0, j=0, k, tmp, arr=this.routes;
 		let matches=[], params={}, handlers=[];
 		for (; i < arr.length; i++) {
 			tmp = arr[i];
-			if (tmp.method.length === 0 || tmp.method === method || isHEAD && tmp.method === 'GET') {
+			if (tmp.mthidx === mthidx  || !tmp.mthidx || (isHEAD && tmp.mthidx===1) ) {
```
Comment (lukeed) [inline index.mjs:53]: The `!tmp.mthidx` is a bug. If a method that isnt in the map is added, it'll have `undefined` index which incorrectly matches this condition.

Comment (Kikobeats) [reply index.mjs:53]: hum I'm not sure how to rewrite it in a way you don't lose the perf you gain

Comment (lukeed) [reply index.mjs:53]: `tmp.mthidx === 0`

**Comment by Kikobeats** [comment]
done; next PR will be to move from Travis to GitHub Actions 😛

**Comment by lukeed** [comment]
Quick bench on Node 18.12.1 because I didn't know what `~10ms` meant in context.

```
# Before
GET / 
  12,864,415 ops/sec ±0.79% (95 runs sampled)
POST /users 
  15,691,754 ops/sec ±0.24% (96 runs sampled)
GET /users/:id 
   9,154,394 ops/sec ±0.15% (98 runs sampled)
PUT /users/:id/books/:title? 
   8,657,366 ops/sec ±0.16% (101 runs sampled)
DELETE /users/:id/books/:title 
   7,996,343 ops/sec ±0.21% (99 runs sampled)
PATCH /^[/]users[/](?<id>\d+)[/]?$/ 
   6,144,565 ops/sec ±0.13% (101 runs sampled)
OPTIONS /^[/]users[/](?<id>\d+)[/]books[/](?<title>[^/]+)[/]?$/ 
   4,634,106 ops/sec ±0.15% (96 runs sampled)
HEAD /hello (all) 
  13,135,000 ops/sec ±0.17% (100 runs sampled)

# After
GET / 
  14,366,336 ops/sec ±0.27% (95 runs sampled)
POST /users 
  16,363,969 ops/sec ±0.13% (98 runs sampled)
GET /users/:id
   9,146,330 ops/sec ±0.18% (95 runs sampled)
PUT /users/:id/books/:title? 
   8,896,106 ops/sec ±0.15% (100 runs sampled)
DELETE /users/:id/books/:title 
   8,133,663 ops/sec ±0.24% (97 runs sampled)
PATCH /^[/]users[/](?<id>\d+)[/]?$/ 
   6,202,659 ops/sec ±0.13% (99 runs sampled)
OPTIONS /^[/]users[/](?<id>\d+)[/]books[/](?<title>[^/]+)[/]?$/ 
   4,715,342 ops/sec ±0.18% (99 runs sampled)
HEAD /hello (all) 
  13,072,490 ops/sec ±0.13% (100 runs sampled)
```

For static matching (first two & last one) there is 4-10% perf increase, if any. Everything else is limited by RegExp matching

Thanks for picking this up~

**Pull request merged**